hta/common/trace_filter.py

```diff
@@ -326,7 +339,17 @@ def __call__(
             logger.warning("DataFrame does not contain a 'stream' column.")
             return df
 
-        return df.loc[df["stream"] == -1]
+        if symbol_table is None:
+            return df.loc[df["stream"] == -1]
```

Comment:
should we have a warning here similar to what you added in this conditional for `GPUKernelFilter`?

Reply:
Just wanted to avoid causing too many warnings. let me add it